starter/app/src/main/java/com/udacity/project4/authentication/AuthenticationActivity.kt

```diff
@@ -1,24 +1,139 @@
 package com.udacity.project4.authentication
 
+import android.app.Activity
+import android.content.Intent
 import android.os.Bundle
+import android.util.Log
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.activity.viewModels
 import androidx.appcompat.app.AppCompatActivity
+import androidx.databinding.DataBindingUtil
+import com.firebase.ui.auth.AuthUI
+import com.firebase.ui.auth.IdpResponse
+import com.google.firebase.auth.FirebaseAuth
 import com.udacity.project4.R
+import com.udacity.project4.databinding.ActivityAuthenticationBinding
+import com.udacity.project4.locationreminders.RemindersActivity
 
 /**
  * This class should be the starting point of the app, It asks the users to sign in / register, and redirects the
  * signed in users to the RemindersActivity.
  */
 class AuthenticationActivity : AppCompatActivity() {
 
+    companion object {
+        const val TAG = "MainFragment"
+    }
+
+    // Get a reference to the ViewModel scoped to this Fragment
+    private val viewModel by viewModels<LoginViewModel>()
+    private lateinit var binding: ActivityAuthenticationBinding
+    private var fakeLogin = true
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_authentication)
-//         TODO: Implement the create account and sign in using FirebaseUI, use sign in using email and sign in using Google
+        val layoutId = R.layout.activity_authentication
+        binding = DataBindingUtil.setContentView(this, layoutId)
+        binding.authButton.text = getString(R.string.login)
+
+        // TODO: Line below was commented to avoid losing time Login In on Firebase
```

Comment:
@graffiti75 I can see you've added new lines, but I can not see any TODOs listed here in this file for the learners!